osclib/stagingapi.py

```diff
@@ -888,21 +888,70 @@ def build_switch_staging_project(self, target_project):
             target_flag = 'enable'
         self.build_switch_prj(target_project, target_flag)
 
-        if self.project_exists(target_project + ":DVD"):
+        if self.item_exists(target_project + ":DVD"):
             self.build_switch_prj(target_project + ":DVD", target_flag)
 
-    def project_exists(self, project):
+    def item_exists(self, project, package=None):
         """
         Return true if the given project exists
         :param project: project name to check
+        :param package: optional package to check
         """
-        url = self.makeurl(['source', project, '_meta'])
+        if package:
+            url = self.makeurl(['source', project, package, '_meta'])
+        else:
+            url = self.makeurl(['source', project, '_meta'])
         try:
             http_GET(url)
         except urllib2.HTTPError:
             return False
         return True
 
+    def package_version(self, project, package):
+        """
+        Return the version of a package, None in case the package does not exist
+        The first non-commented Version: tag found is used.
+        :param project: the project the package resides in
+        :param package: the package to check
+        :param product: if passed, the package to be checked is considered to be part of _product
+        """
+        if not self.item_exists(project, package):
+            return None
+
+        version = None
+
+        specfile = self.api.load_file_content(project, package, '{}.spec'.format(package))
+        if specfile:
+            try:
+                version = re.findall("Version:(.*)",specfile,re.MULTILINE)[0].strip()
+            except:
```

Comment:
Good catch... Or at least it must not be behind a # (which is implicit if it is at the beginning of a line).

Will update the PR in a bit

Sent from my iPad

> On 20 Jan 2015, at 08:12, Stephan Kulow <notifications@github.com> wrote:
> 
> In osclib/stagingapi.py:
> 
> > +        Return the version of a package, None in case the package does not exist
> > +        The first non-commented Version: tag found is used.
> > +        :param project: the project the package resides in
> > +        :param package: the package to check
> > +        :param product: if passed, the package to be checked is considered to be part of _product
> > +        """
> > +        if not self.item_exists(project, package):
> > +            return None
> > +
> > +        version = None
> > +
> > +        specfile = self.api.load_file_content(project, package, '{}.spec'.format(package))
> > +        if specfile:
> > +            try:
> > +                version = re.findall("Version:(.*)",specfile,re.MULTILINE)[0].strip()
> > +            except:
> not sure how findall works, but you want to anchor the Version: at line start.
> 
> —
> Reply to this email directly or view it on GitHub.
>